src/boot/efi/stub.c

```diff
@@ -159,23 +162,74 @@ static void export_stub_variables(EFI_LOADED_IMAGE_PROTOCOL *loaded_image) {
         (void) efivar_set_str16(MAKE_GUID_PTR(LOADER), u"StubInfo", u"systemd-stub " GIT_VERSION, 0);
 
         (void) efivar_set_uint64_le(MAKE_GUID_PTR(LOADER), u"StubFeatures", stub_features, 0);
+
+        (void) efivar_set_uint64_str16(MAKE_GUID_PTR(LOADER), u"StubProfile", profile, 0);
+}
+
+static bool parse_profile_from_cmdline(char16_t **cmdline, unsigned *ret_profile) {
+        assert(cmdline);
+        assert(*cmdline);
+        assert(ret_profile);
+
+        const char16_t *p = *cmdline;
+        if (p[0] != '@')
+                goto nothing;
+
+        uint64_t u;
+        const char16_t *tail;
+        if (!parse_number16(p + 1, &u, &tail))
+                goto nothing;
+        if (u > UINT_MAX)
+                goto nothing;
+        if (tail[0] == u' ')
+                tail++;
```

Comment:
You could have more than one space, I'd either strip all of them or none, but doing just one seems a bit iffy

Reply:
this is actually on purpose. THis is not about "cleaning up" the string or so, but just to strip off the minimum profile suffix that makes sense to strip. After all the result goes into mesaurement, and hence any additional mangling we do here would have to be replicated by any tool that wants to predict measurements.

I'll add a comment about this.